ivy/functional/ivy/manipulation.py

```diff
@@ -1444,3 +1444,35 @@ def zero_pad(
 
     """
     return current_backend(x).zero_pad(x, pad_width, out=out)
+
+
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_nestable
+@handle_exceptions
+def fill_diagonal(
```

Comment:
I think that this function should be put in ivy//ivy/functional/extensions/manipulation.py instead of here.

Reply:
No problem, I can fix that. I assume this means everything else is looking OK so far?

Reply:
Just in case you did not see that, you also got some lint formatting error to fix (in other files).